File: ruby/bob/bob.rb
module BookKeeping
  VERSION = 1
end

# bob the builder
class Bob
  RESPONSE = {
    question: 'Sure.',
    yell: 'Whoa, chill out!',
    silence: 'Fine. Be that way!',
    default: 'Whatever.'
  }.freeze

  def self.hey(remark)
    RESPONSE[type_of(remark)]
  end

  def self.type_of(remark)
    if remark.upcase == remark && remark.downcase != remark
      :yell
    elsif remark.strip.end_with?('?')
      :question
    elsif remark.strip.empty?
      :silence
    else
      :default
    end
  end
end

File: elixir/bob/bob.exs
defmodule Bob do
  def hey(input) do
    case check(input) do
        :shouting -> "Whoa, chill out!"
        :question -> "Sure."
        :silence -> "Fine. Be that way!"
        _ -> "Whatever."

    end
  end

  defp check(input) do
    cond do
      String.ends_with?(input,"?") -> :question
      String.trim(input) == "" -> :silence
      shouting?(input) -> :shouting
      true -> :default
    end
  end

  defp shouting?(input) do
     String.upcase(input) == input && String.downcase(input) != input
  end
end
